### ISSUE TITLE
Allow for routing of different sections of the application?

### ISSUE THREAD

**Issue by JScreen** [Flatline4/Flatline4#8]
Essentially, implement something similar to Partials in ASP.NET MVC, where you can define different parts of the application separately, name them, and then include them in other places within the application. This is currently supported, but requires making a function that acts like a child to an already-parent object (See Demo/Navbar.js) 

It would be better if we had something in place that also allowed us to "route" different parts of the application. Ex: say you just want one of the side panels to change individually of the main content? 

What do you think? 👍 ?

**Comment by BRKurek** [comment]
Yeah, I think this is definitely something we should support. Once we get away from using the body tag as the start of the HTML generation this shouldn't be too hard to implement.

**Comment by pranaygp** [comment]
@BRKurek makes a good point
